src/instructlab/configuration.py

```diff
@@ -381,55 +419,97 @@ def get_model_family(forced, model_path):
     return guess if guess in MODEL_FAMILIES else DEFAULT_MODEL_FAMILY
 
 
+def ensure_storage_directories_exist():
+    """
+    Ensures that the default directories used by ilab exist.
+    """
+    os.makedirs(ILAB_CONFIG_HOME, exist_ok=True, mode=0o755)
+
+    for storage_dir in [
+        STORAGE_DIR_NAMES.CHECKPOINTS,
+        STORAGE_DIR_NAMES.MODELS,
+        STORAGE_DIR_NAMES.DATASETS,
+        STORAGE_DIR_NAMES.TAXONOMY,
+        STORAGE_DIR_NAMES.INTERNAL,
+    ]:
+        os.makedirs(
+            os.path.join(ILAB_DATA_HOME, storage_dir), exist_ok=True, mode=0o755
```

Comment:
Is there any particular reason why you are overriding the user's umask?

Reply:
What would you recommend instead? I was under the impression that `0o755` would be fine for this.

Reply:
Based on the [documentation of mkdir](https://docs.python.org/3/library/os.html#os.mkdir) (what gets used under the hood of `os.makedirs`), it seems like the umask is subtracted away from the mode we set.
My understanding is that if I'm setting `0o755` and the umask is `0077`, then the resulting permissions will still be `0700`. 

Maybe I'm not understanding something, but I don't believe we are overriding the user's umask here.
Please correct me if I'm wrong here.